src/main/scala/util/RunUtils.scala

```diff
@@ -931,6 +981,8 @@ object RunUtils {
         result.reachingDefs,
         ctx.program
       )
+      Logger.info("[!] Generating Function Summaries")
+      modified = modified | IRTransform.generateFunctionSummaries(ctx, ctx.program, result.constPropResult, result.varDepsSummaries)
```

Comment:
I don't think it makes sense to update `modified` here, at least at the moment. The procedure summaries don't have any impact (and can't have any impact) on any of the other analyses right now so there isn't any need to do another iteration.